lib/puppet/type/xml_fragment.rb

```diff
@@ -57,8 +58,6 @@
           unless val.is_a?(String)
             value['attributes'][key] = val.to_s
           end
-
-          raise ArgumentError, "Attribute #{key} must be a string." unless value['attributes'][key].is_a?(String)
```

Comment:
I think the conversion to a string above means that this validation could never fail..